optimum/habana/transformers/models/mpt/modeling_mpt.py

```diff
@@ -305,6 +307,13 @@ def prepare_inputs_for_generation(
                 input_ids = input_ids[:, remove_prefix_length:]
             else:
                 input_ids = torch.index_select(input_ids, 1, token_idx - 1)
+            # Converting back to tuples as it should be, so there's no type mismatch when calling graph
+            past_key_values = tuple([tuple(kv) for kv in past_key_values])
+        elif bucket_internal and token_idx is not None:
```

Comment:
```suggestion
        elif all([bucket_internal, token_idx]):
```

Suggestion to use all to make sure all variables are truthy (i.e., not None, not an empty string, not zero, etc.)